commit 535d6a9c2db49df90968da9f292cb41605c05efe
Author: Benjamin Gilbert <bgilbert@backtick.net>
Date:   Thu Jan 5 20:29:00 2023 -0500

    meson: don't dllexport when built as static library
    
    If a static Pixman is linked with a dynamic library, Pixman shouldn't
    export its own symbols into the latter's ABI.

diff --git a/pixman/meson.build b/pixman/meson.build
index 652cda3418e2..78b37b95bfff 100644
--- a/pixman/meson.build
+++ b/pixman/meson.build
@@ -31,7 +31,8 @@ version_h = configure_file(
 )
 
 libpixman_extra_cargs = []
-if cc.has_function_attribute('dllexport')
+default_library = get_option('default_library')
+if cc.has_function_attribute('dllexport') and default_library != 'static'
   libpixman_extra_cargs = ['-DPIXMAN_API=__declspec(dllexport)']
 endif
 
